packages/api/tests/lambdas/sf-event-sqs-to-db-records/test-write-execution.js

```diff
@@ -308,3 +310,27 @@ test.serial('writeExecution() does not persist records to Dynamo or RDS if RDS w
   t.false(await executionModel.exists({ arn: executionArn }));
   t.false(await executionPgModel.exists(knex, { arn: executionArn }));
 });
+
+test.serial('writeExecution() correctly sets both original_payload and final_payload in postgres when execution records are run in sequence', async (t) => {
+  const {
+    cumulusMessage,
+    knex,
+    executionArn,
+    executionPgModel,
+  } = t.context;
+
+  const originalPayload = { original: 'payload' };
+  const finalPayload = { final: 'payload' };
+
+  cumulusMessage.meta.status = 'running';
+  cumulusMessage.payload = originalPayload;
+  await writeExecution({ cumulusMessage, knex });
+
+  cumulusMessage.meta.status = 'completed';
+  cumulusMessage.payload = finalPayload;
+  await writeExecution({ cumulusMessage, knex });
```

Comment:
There is nothing wrong with the way you've set up your test but I am curious about the advantages of writing it the way you did versus using `Promise.all` 😯

Reply:
Ah, sorry, I somehow thought you were running them in parallel as opposed to in sequence. With your setup, `await writeExections` aren't necessarily running in sequence, right?

Reply:
In fact we *explicitly* want to run them in that order - the nature of this failure is due to the sequence in which an execution is written as runnning, then updated as the same completed execution.       This test is the result of TDDing the success case for the bug for #2275  -- prior to the update original_payload/originalPayload was being removed when an execution changed to any state other than running, as knex takes { key: undefined } on upsert as a reason to remove an existing value.